Sources/Architecture/ARM/SoftwareBreakpointManager.cpp

```diff
@@ -35,7 +35,10 @@ void SoftwareBreakpointManager::clear() {
 }
 
 ErrorCode SoftwareBreakpointManager::add(Address const &address, Type type,
-                                         size_t size) {
+                                         size_t size, Mode mode) {
+  if (mode != kModeExec)
```

Comment:
Also, I looked at software watchpoints a little bit, and they are handled by the client directly. The debug server has nothing to do with them, and the protocol doesn't support setting "software watchpoints". PR#268 should make us safe to put an assertion there.